js/checkboxgroup/checkboxgroup.test.ts

```diff
@@ -0,0 +1,345 @@
+import { test, describe, assert, afterEach, vi } from "vitest";
+import { cleanup, render } from "@gradio/tootils";
+import event from "@testing-library/user-event";
+import { setupi18n } from "../app/src/i18n";
+
+import CheckboxGroup from "./Index.svelte";
+import type { LoadingStatus } from "@gradio/statustracker";
+
+const loading_status: LoadingStatus = {
+	eta: 0,
+	queue_position: 1,
+	queue_size: 1,
+	status: "complete" as LoadingStatus["status"],
+	scroll_to_output: false,
+	visible: true,
+	fn_index: 0,
+	show_progress: "full"
+};
+
+beforeEach(() => {
+	setupi18n();
+});
+
+afterEach(cleanup);
+
+describe("Values", () => {
```

Comment:
I use describe blocks to make it clear what a series of tests are testing.

There is no value in a suite call `checkboxgroup.test.ts` having a describe block of `CheckboxGroup`.

In this suite we are focused on the value being updated correctly both in the UI and the actual `value` of the component. This is very easy to test multiple variants. Here i tested checking things of worked + updated the value/ checked status of the input in various forms.

I also checked that setting + updating the value from outside works as expected. They are different code paths so it is important to test them separately.